### PR TITLE
Implementation of terminal tabs

### PR DESCRIPTION
**Pull request by Yaraslaut**
Closes https://github.com/contour-terminal/contour/issues/90

### Checklist

- [x] documentation
- [x] create new tab
- [x] switch to tab on the left
- [x] switch to tab on the right
- [x] switch to tab by index
- [x] close tab
- [x] ensure window sizing is not affected by tab spawning or tab switching

### CODE REVIEW

File: src/contour/ui.template/main.qml.in [hunk 1]
```diff
@@ -36,6 +36,21 @@ ApplicationWindow
...
+        width: parent.width
+        TabButton {
+            text: qsTr("1")
+        }
+        TabButton {
+            text: qsTr("2")
+        }
+        onCurrentIndexChanged: {
+            print("currentIndex changed to", currentIndex)
+            terminalSessions.setSession(currentIndex)
```
Comment (christianparpart) [inline src/contour/ui.template/main.qml.in:50]: oh wow. THAT is a really nice take on tabs! :)

**Review by christianparpart — APPROVED** [review]
Thank you, @Yaraslaut.

**Comment by denisgolius** [comment]
Hey @christianparpart , when to expect the new release? :)

**Comment by Yaraslaut** [comment]
> Hey @christianparpart , when to expect the new release? :)

hi @denisgolius, if you want to try out tabs asap, you can build from master, otherwise it will take some time until we have some confidence that tabs are somewhat stable, and we want to add some customization into it as well before the next release. 
At the moment we are not planning to create new release, so most likely later this year

**Comment by denisgolius** [comment]
> > Hey @christianparpart , when to expect the new release? :)
> 
> hi @denisgolius, if you want to try out tabs asap, you can build from master, otherwise it will take some time until we have some confidence that tabs are somewhat stable, and we want to add some customization into it as well before the next release. At the moment we are not planning to create new release, so most likely later this year

Ok, thanks!

**Comment by christianparpart** [comment]
Thanks for your interest. Before a bad experience may make people unhappy, we want to make sure that this impactful change is not causing any troubles. A few are known (pending PR) and one is left out to be fixed apart from what @Yaraslaut also already said, please bear with us, and submit bug tickets whenever you find something and we will try our best to improve based on the feedback you guys are giving us.

**Pull request merged**